src/services/Frontline.js

```diff
@@ -124,6 +135,111 @@ class Frontline {
       }
     }
   }
+
+  /**
+   * Load Infinity
+   */
+
+  httpRequestWrapper = (method, URL) => {
+    return new Promise((resolve, reject) => {
+      const xhr_obj = new XMLHttpRequest();
+      xhr_obj.responseType = "json";
+      xhr_obj.open(method, URL);
+      xhr_obj.onload = () => {
+        const data = xhr_obj.response;
+        resolve(data);
+      };
+      xhr_obj.onerror = () => {
+        reject("failed");
+      };
+      xhr_obj.send();
+    });
+  };
+
+  async getProducts(apiEndpoint, params) {
+    return await this.httpRequestWrapper(
+      "GET",
+      `${apiEndpoint}/products.json?${params}`
+    );
+  }
+
+  footerHeight = () => {
+    var sum = 0;
+    LOAD_INFINITY_CLASSES.footer.split(",").forEach((className) => {
```

Comment:
Why are multiple selectors used here? And why do you have to sum all the individual heights together?

Reply:
In the `wundery` theme, we are having only 1 element as the footer which is `.footer-wrapper`.
But in the `speedy` theme, we have some other elements for the footer which are `.heading-cat` and `.homedescription` also. So I need to sum up all footer elements' heights to make it call the paginated request at the correct position.

Reply:
The purpose of this footer selector just seems to be to detect if scrolling reached the end of the product list. So basically for answering the question `has the next element become visible?`

To not go down a rabbit hole of supporting all sorts of `what's the next element after products` cases ... instead of asking for the footer you could just query the position and size of the product list (`LOAD_INFINITY_CLASSES.appendProducts`) - using `top + height` of the result of `getBoundingClientRect()` for this selector should give you the exact result (scroll position of the bottom edge of the product list) you need.

As an alternative, you could also query the position of the `loadMore` selector - but this would rely on the `loadMore` element to appear directly after the product list. This is however how most infinite loading implementations do it: the add a "scroll spy" element which marks the end of the list. This element is then watched on scroll using an [Intersection observer](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API). 

This intersection observer approach is also much more efficient because of the following: the `node.offsetHeight` and `node.getBoundingClientRect` have to do a full layout reflow to determine their values. So if you manually observe the `scroll` event you are observing a rapidly firing event and on each event occurence, force the browser to reflow the whole document (which is basically the most expensive operation you can do).

Reply:
Thanks for this amazing solution!